### DISCUSSION TITLE
help wanted: eglot not enabling yasnippet despite yasnippet being available

### DISCUSSION THREAD

**Discussion by rileyrg** [joaotavora/eglot#1472] [General]
Emacs from source 30.0.92
My understanding is that eglot should enable yasnippet insertion into completions if yasnippet is installed. 

This isnt happening for me with cpp buffers and clangd. I need to specifically call yas-minor-mode in my c++-ts-mode-hook in order to see yasnippets like "ffo" - and then they don't appear in the completion popup - rather I need to C-g from the completion popup and then hit TAB. I know it sounds confusing - it is . If I DONT self enable yas-minor-mode I see "snippets" in the completion popup but they dont seem to correspond to whats in my yasnippets directory for c++ mode (eg no ffo).  Where are they coming from?  

Is there anything obvious I might be missing?

Note that yasnippet is difinitely loaded before any eglot buffers are opened. Originally I was using corfu but I'm now using "company" to more toe the company line... excuse the pun.. and use the more supported completion mechanism. eglot is enabled in my cpp buffers using a call to eglot-ensure added to the c++-ts-mode-hook.

**Comment by rileyrg** [comment]
This function is called, but (yas-minor-mode 1) isnt applied to my buffer. 

```
(defun eglot--snippet-expansion-fn ()
  "Compute a function to expand snippets.
Doubles as an indicator of snippet support."
  (and (fboundp 'yas-minor-mode)
       (lambda (&rest args)
         (with-no-warnings
           (unless (bound-and-true-p yas-minor-mode) (yas-minor-mode 1))
           (apply #'yas-expand-snippet args)))))
```

**Comment by manuel-uberti** [comment]
I suggest going through the [Reporting bugs](https://joaotavora.github.io/eglot/#Reporting-bugs-1) section of the manual in order to share as many useful information as possible to understand the problem.
